File: ios/SpentyAI/Core/Models/FeatureRequest.swift
import Foundation

struct FeatureRequest: Codable, Identifiable, Hashable {
    var id: String { requestId }

    let requestId: String
    var title: String
    var description: String?
    var category: String?
    var status: String?
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case requestId = "request_id"
        case title
        case description
        case category
        case status
        case createdAt = "created_at"
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/FeatureRequest.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class FeatureRequest(
    @SerialName("request_id") val id: String,
    val title: String,
    val description: String,
    val status: FeatureRequestStatus = FeatureRequestStatus.OPEN,
    val votes: Int = 0,
    @SerialName("has_voted") val hasVoted: Boolean = false,
    val category: String? = null,
    @SerialName("submitted_by") val submittedBy: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)

@Serializable
enum class FeatureRequestStatus {
    @SerialName("open") OPEN,
    @SerialName("planned") PLANNED,
    @SerialName("in_progress") IN_PROGRESS,
    @SerialName("completed") COMPLETED,
    @SerialName("declined") DECLINED
}
